### PR TITLE
chore: migrate cache plugin from the integrations repo

### LINKED ISSUE #1: Migrate implementation from integrations to this repo

### LINKED ISSUE #2: Cache plugin should use msgpack type for storing value

### LINKED ISSUE #170: Cache plugin implementation

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
- Closes: #1 
- Closes: #2 
- Closes: polywrap/python-client#170

### CODE REVIEW

File: implementations/in-memory/js/src/wrap/wrap.info.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+          "scalar": {
+            "kind": 4,
+            "name": "set",
+            "required": true,
+            "type": "Boolean"
+          },
+          "type": "Boolean"
+        },
+        "type": "Method"
+      },
```
Comment (Niraj-Kamdar) [inline implementations/in-memory/js/src/wrap/wrap.info.ts]: Ignore this

File: implementations/in-memory/js/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    this._cache.set(args.key, cacheItem);
+    if (expirationTime !== null) {
+      this._expirationHeap.push({ key: args.key, expirationTime });
+    }
+
+    return true;
+  }
+
+  public add(args: Args_add): boolean {
+    if (args.timeout) return false; // Timeout not implemented!
```
Comment (krisbitney) [inline implementations/in-memory/js/src/index.ts]: Won't the `set` method handle the timeout logic?

**Review by krisbitney — APPROVED** [review]
LGTM

File: implementations/in-memory/js/src/index.ts (hunk 2 above)
Comment (Niraj-Kamdar) [reply implementations/in-memory/js/src/index.ts]: oops! that's previous implementation leftover! Nice catch!

**Pull request merged**